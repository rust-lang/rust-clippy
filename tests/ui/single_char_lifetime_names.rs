#![warn(clippy::single_char_lifetime_names)]

struct DiagnosticCtx<'a, 'b>
where
    'a: 'b,
{
    _source: &'a str,
    _unit: &'b (),
}

impl<'a, 'b> DiagnosticCtx<'a, 'b> {
    fn new(source: &'a str, unit: &'b ()) -> DiagnosticCtx<'a, 'b> {
        Self {
            _source: source,
            _unit: unit,
        }
    }
}

impl<'src, 'unit> DiagnosticCtx<'src, 'unit> {
    fn new_pass(source: &'src str, unit: &'unit ()) -> DiagnosticCtx<'src, 'unit> {
        Self {
            _source: source,
            _unit: unit,
        }
    }
}

fn main() {
    let src = "loop {}";
    let unit = ();
    DiagnosticCtx::new(src, &unit);
}
