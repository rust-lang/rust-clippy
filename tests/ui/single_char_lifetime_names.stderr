error: single-character lifetime names are likely uninformative
  --> $DIR/single_char_lifetime_names.rs:5:22
   |
LL | struct DiagnosticCtx<'a, 'b>
   |                      ^^
   |
   = note: `-D clippy::single-char-lifetime-names` implied by `-D warnings`
   = help: use a more informative name

error: single-character lifetime names are likely uninformative
  --> $DIR/single_char_lifetime_names.rs:5:26
   |
LL | struct DiagnosticCtx<'a, 'b>
   |                          ^^
   |
   = help: use a more informative name

error: single-character lifetime names are likely uninformative
  --> $DIR/single_char_lifetime_names.rs:14:6
   |
LL | impl<'a, 'b> DiagnosticCtx<'a, 'b> {
   |      ^^
   |
   = help: use a more informative name

error: single-character lifetime names are likely uninformative
  --> $DIR/single_char_lifetime_names.rs:14:10
   |
LL | impl<'a, 'b> DiagnosticCtx<'a, 'b> {
   |          ^^
   |
   = help: use a more informative name

error: single-character lifetime names are likely uninformative
  --> $DIR/single_char_lifetime_names.rs:34:15
   |
LL | fn split_once<'a>(base: &'a str, other: &'_ str) -> (&'a str, Option<&'a str>) {
   |               ^^
   |
   = help: use a more informative name

error: aborting due to 5 previous errors

